dotnet/src/SemanticKernel.Core/Planning/Plan.cs

```diff
@@ -50,6 +50,12 @@ public sealed class Plan : IPlan
     [JsonPropertyName("outputs")]
     public IList<string> Outputs { get; set; } = new List<string>();
 
+    /// <summary>
+    /// Collection of step results
+    /// </summary>
+    [JsonIgnore]
+    public IList<FunctionResult> StepResults { get; set; } = new List<FunctionResult>();
```

Comment:
Not fully sure about exposing this new property.

Plan has the "Object Model", "Execution", "State" and now the "Result" responsibity. :)